langchain/src/chat_models/azure_ml.ts

```diff
@@ -0,0 +1,142 @@
+   * @param modelArgs A dictionary of model arguments.
+   * @returns A string representing the formatted request payload.
+   */
+  formatRequestPayload: (
+    messages: BaseMessage[],
+    modelArgs: Record<string, unknown>
+  ) => string;
+  /**
+   * Formats the response payload from the AzureML endpoint. It takes a
+   * response payload as input and returns a string representing the
+   * formatted response.
+   * @param responsePayload The response payload from the AzureML endpoint.
+   * @returns A string representing the formatted response.
+   */
+  formatResponsePayload: (output: string) => string;
+}
+
+export class LlamaContentFormatter implements ChatContentFormatter {
+  _convertMessageToRecord(message: BaseMessage): Record<string, unknown> {
+    if (message._getType() === "human") {
+      return { role: "user", content: message.content };
+    } else if (message._getType() === "ai") {
+      return { role: "assistant", content: message.content };
+    } else {
+      return { role: message._getType(), content: message.content };
+    }
+  }
+
+  formatRequestPayload(
+    messages: BaseMessage[],
+    modelArgs: Record<string, unknown>
+  ): string {
+    let msgs = messages.map((message) => {
+      this._convertMessageToRecord(message);
+    });
+    return JSON.stringify({
+      input_data: {
+        input_string: msgs,
+        parameters: modelArgs,
+      },
+    });
+  }
+
+  formatResponsePayload(responsePayload: string) {
+    const response = JSON.parse(responsePayload);
+    return response.output;
+  }
+}
+
+/**
+ * Type definition for the input parameters of the AzureMLChatOnlineEndpoint class.
+ */
+export interface AzureMLChatParams extends BaseChatModelParams {
+  endpointUrl?: string;
+  endpointApiKey?: string;
+  modelArgs?: Record<string, unknown>;
+  contentFormatter?: ChatContentFormatter;
+}
+
+/**
+ * Class that represents the chat model. It extends the SimpleChatModel class and implements the AzureMLChatInput interface.
+ */
+export class AzureMLChatOnlineEndpoint
+  extends SimpleChatModel
+  implements AzureMLChatParams
+{
+  static lc_name() {
+    return "AzureMLChatOnlineEndpoint";
+  }
+  static lc_description() {
+    return "A class for interacting with AzureML Chat models.";
+  }
+  endpointUrl: string;
+  endpointApiKey: string;
+  modelArgs?: Record<string, unknown>;
+  contentFormatter: ChatContentFormatter;
+  httpClient: AzureMLHttpClient;
+
+  constructor(fields: AzureMLChatParams) {
+    super(fields ?? {});
+    if (!fields?.endpointUrl && !getEnvironmentVariable("AZUREML_URL")) {
+      throw new Error("No Azure ML Url found.");
+    }
+    if (!fields?.endpointApiKey && !getEnvironmentVariable("AZUREML_API_KEY")) {
+      throw new Error("No Azure ML ApiKey found.");
+    }
+    if (!fields?.contentFormatter) {
+      throw new Error("No Content Formatter provided.");
+    }
+
+    this.endpointUrl =
+      fields.endpointUrl || getEnvironmentVariable("AZUREML_URL") + "";
```

Comment:
Prefer `??`. Please change for the rest. Also why is there `+ ""`?
```suggestion
      fields.endpointUrl ?? getEnvironmentVariable("AZUREML_URL") + "";
```